File: Classes/AMGAboutHeaderView.m
//
//  AMGAboutHeaderView.m
//
//  Created by Vincent Tourraine on 11/03/2019.
//  Copyright © 2019 Studio AMANgA. All rights reserved.
//

#import "AMGAboutHeaderView.h"

#import <NGAParallaxMotion.h>

@implementation AMGAboutHeaderView

- (instancetype)initWithIconImageNamed:(NSString *)iconImageName {
    self = [super initWithFrame:CGRectMake(0, 0, 320, 200)];
    if (self) {
        UIImage *image = [UIImage imageNamed:iconImageName];

        UIImageView *iconImageView = [[UIImageView alloc] initWithImage:image];
        iconImageView.contentMode = UIViewContentModeCenter;
        iconImageView.center = self.center;
        iconImageView.autoresizingMask = UIViewAutoresizingFlexibleLeftMargin | UIViewAutoresizingFlexibleRightMargin;
        iconImageView.clipsToBounds = YES;
        iconImageView.layer.cornerRadius = 25;
        iconImageView.parallaxIntensity = 40;

        [self addSubview:iconImageView];
    }

    return self;
}

@end

File: Sources/AMGAboutViewController/AMGAboutHeaderView.swift
//
//  AMGAboutHeaderView.swift
//
//  Created by Vincent Tourraine on 11/03/2019.
//  Copyright © 2019-2022 Studio AMANgA. All rights reserved.
//

#if !os(macOS)

import UIKit

class AMGAboutHeaderView: UIView {

    convenience init(iconImageNamed iconImageName: String) {
        self.init(frame: CGRect(x: 0, y: 0, width: 320, height: 200))

        guard let image = UIImage(named: iconImageName) else {
            return
        }

        let iconImageView = UIImageView(image: image)
        iconImageView.contentMode = .center
        iconImageView.center = center
        iconImageView.autoresizingMask = [.flexibleLeftMargin, .flexibleRightMargin]
        iconImageView.clipsToBounds = true
        iconImageView.layer.cornerRadius = 25
        iconImageView.layer.cornerCurve = .continuous
        addSubview(iconImageView)
    }
}

#endif
